commands/onboard.js

```diff
@@ -1,19 +1,59 @@
+import readline from 'node:readline/promises'
+
 /**
  * Onboards a user to an organization.
  * @param {{ client: import('../github-api.js').default, logger: import('pino').Logger }} deps
  * @param {{ org: string, username: string, dryRun: boolean }} options
  * @returns {Promise<void>}
  */
-export default async function onboard ({ client, logger }, { org, username, dryRun }) {
-  const orgId = await client.getOrgId(org)
-  logger.info('Organization ID %s', orgId)
+export default async function onboard ({ client, logger }, { org, username, joiningTeams, dryRun }) {
+  const joiningUser = await client.getUserInfo(username)
+  if (!await confirm(`Are you sure you want to onboard ${joiningUser.login} [${joiningUser.name}] to ${org}?`)) {
+    logger.warn('Aborting onboarding')
+    process.exit(0)
+  }
 
-  const orgChart = await client.getOrgChart(org)
+  const orgData = await client.getOrgData(org)
+  logger.info('Organization ID %s', orgData.id)
+
+  const orgTeams = await client.getOrgChart(orgData)
+  const destinationTeams = orgTeams.filter(t => joiningTeams.includes(t.slug))
+  if (destinationTeams.length !== joiningTeams.length) {
+    const missing = joiningTeams.filter(t => destinationTeams.find(dt => dt.slug === t) == null)
```

Comment:
What if you use the Set difference?
If you make `joiningTeams` a set, you can also use it on line 20, to replace the includes with the has.